source/funkin/editors/character/CharacterEditor.hx

```diff
@@ -0,0 +1,696 @@
+					new FlxPoint(char.characterData.animations[curAnim].x, char.characterData.animations[curAnim].y));
+
+				char.addOffset(char.characterData.animations[curAnim].anim, char.characterData.animations[curAnim].x,
+					char.characterData.animations[curAnim].y);
+				ghostChar.addOffset(char.characterData.animations[curAnim].anim, char.characterData.animations[curAnim].x,
+					char.characterData.animations[curAnim].y);
+				/*ghostChar.animOffsets.set(char.animation.curAnim.name,
+					new FlxPoint(char.characterData.animations[curAnim].x, char.characterData.animations[curAnim].y)); */
+
+				char.playAnim(char.characterData.animations[curAnim].name, false);
+				if (ghostChar.animation.curAnim != null
+					&& char.animation.curAnim != null
+					&& char.animation.curAnim.name == ghostChar.animation.curAnim.name)
+				{
+					ghostChar.playAnim(char.animation.curAnim.name, false);
+				}
+			}
+		}
+	}
+
+	function controlCamera(elapsed:Float)
+	{
+		var ofs:Float = 500 * elapsed;
+		var shift:Bool = (FlxG.keys.pressed.SHIFT);
+		var cam_up:Bool = (FlxG.keys.pressed.W);
+		var cam_down:Bool = (FlxG.keys.pressed.S);
+		var cam_left:Bool = (FlxG.keys.pressed.A);
+		var cam_right:Bool = (FlxG.keys.pressed.D);
+		var cam_zoomers:Array<Bool> = [FlxG.keys.pressed.I, FlxG.keys.pressed.O];
+
+		if (shift)
+			ofs *= 4;
+
+		if (cam_up)
+			camFollow.y -= ofs;
+		if (cam_down)
+			camFollow.y += ofs;
+		if (cam_left)
+			camFollow.x -= ofs;
+		if (cam_right)
+			camFollow.x += ofs;
+
+		var zoom_ofs = FlxG.camera.zoom * elapsed;
+		if (cam_zoomers[0] && camZoom < 3)
+		{
+			camZoom += zoom_ofs;
+			if (camZoom > 3)
+				camZoom = 3;
+		}
+		if (cam_zoomers[1] && camZoom > 0.1)
+		{
+			camZoom -= zoom_ofs;
+			if (camZoom < 0.1)
+				camZoom = 0.1;
+		}
+	}
+
+	function changeAnim()
+	{
+		var space = (FlxG.keys.justPressed.SPACE);
+		var forward = (FlxG.keys.justPressed.L);
+		var backwards = (FlxG.keys.justPressed.K);
+
+		var changelimit = char.characterData.animations.length;
+
+		if (forward || backwards)
+		{
+			if (forward)
+				curAnim += 1;
+			else if (backwards)
+				curAnim -= 1;
+
+			if (curAnim >= changelimit)
+				curAnim = 0;
+			else if (curAnim < 0)
+				curAnim = changelimit - 1;
+
+			char.playAnim(char.characterData.animations[curAnim].name, false);
+			playingAnim.text = 'Animation:\n${char.characterData.animations[curAnim].name}';
+		}
+
+		if (space)
+			char.playAnim(char.characterData.animations[curAnim].name, true);
+
+		char.updateHitbox();
+		ghostChar.updateHitbox();
+	}
+
+	// Animation Windows
+	function add_anim_window(_)
+		FlxG.state.openSubState(new AddAnimScreen(char, ghostChar, this));
+
+	function update_anim_window(_)
+		FlxG.state.openSubState(new SelectAnimUpdate(char, ghostChar, this));
+
+	function remove_anim_window(_)
+		FlxG.state.openSubState(new DeleteAnimScreen(char, ghostChar, this));
+
+	function help_controls_window(_)
+		FlxG.state.openSubState(new HelpControls());
```

Comment:
Why are you using `FlxG.state.openSubState`, when you can just do `openSubState`

Reply:
Didn't know that this was also possible, haha.